### PR TITLE
[Docs] Logging API and configuration

### LINKED ISSUE #5896: Logging API needs to update Documentation
The logging API is missing docs

### PR DESCRIPTION
**Pull request by suyograo**
Fixes #5896

### CODE REVIEW

**Comment by talevy** [comment]
I think we need more structure for these and maybe also add explanations of how to fetch these `path.logs` files. That being said, I think this is perfect and plenty for beta1, we can iterate for GA

LGTM

File: docs/static/logging.asciidoc [hunk 1]
```diff
@@ -0,0 +1,69 @@
+[[logging]]
+== Logstash's Internal Logging
+
+Logstash emits internal logs during it's operation, which are placed in `LS_HOME`/logs. The default logging level is `INFO`.
```
Comment (dedemorton) [inline docs/static/logging.asciidoc]: typo: should be "its operation". Also, I think you might want the back ticks around the whole path `LS_HOME/logs`

File: docs/static/logging.asciidoc [hunk 2]
```diff
@@ -0,0 +1,69 @@
+[[logging]]
+== Logstash's Internal Logging
+
+Logstash emits internal logs during it's operation, which are placed in `LS_HOME`/logs. The default logging level is `INFO`. 
+Logstash's logging framework is based on http://logging.apache.org/log4j/2.x/[Log4j2 framework], and many of it's functionality are exposed directly
```
Comment (dedemorton) [inline docs/static/logging.asciidoc]: Should be "its functionality"

File: docs/static/logging.asciidoc [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+When debugging problems, particularly problems with plugins, it can be helpful to increase the logging level to `DEBUG` 
+to emit more verbose messages. Previously, you could only set a log level which applied to the entire Logstash product. 
+Starting with 5.0, you can configure logging for a particular subsystem in Logstash. For example, if you are 
+debugging issues with Elasticsearch Output, you can increase log levels just for that component. This way 
+you can reduce noise due to excessive logging and focus on the problem area effectively.
+
+=== Log file location
+
+You can specify log file location using `--path.logs` setting.
```
Comment (dedemorton) [inline docs/static/logging.asciidoc]: "specify the log file location"

File: docs/static/logging.asciidoc [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+=== Log4j2 Configuration
+
+Logstash ships with a `log4j2.properties` file with out of the box settings. Users can modify this file directly to change the 
+rotation policy, type and other https://logging.apache.org/log4j/2.x/manual/configuration.html#Loggers[log4j2 configuration]. 
+Any change made to this file is only effective after a process restart.
+
+=== Logging APIs
+
+You could modify the `log4j2.properties` file and restart your Logstash, but that is both tedious and leads to unnecessary 
+downtime. Instead, you can dynamically update logging levels through the a settings API. These settings are effective
```
Comment (dedemorton) [inline docs/static/logging.asciidoc]: "through the settings API" (remove "a")

File: docs/static/logging.asciidoc [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+Logstash ships with a `log4j2.properties` file with out of the box settings. Users can modify this file directly to change the 
+rotation policy, type and other https://logging.apache.org/log4j/2.x/manual/configuration.html#Loggers[log4j2 configuration]. 
+Any change made to this file is only effective after a process restart.
+
+=== Logging APIs
+
+You could modify the `log4j2.properties` file and restart your Logstash, but that is both tedious and leads to unnecessary 
+downtime. Instead, you can dynamically update logging levels through the a settings API. These settings are effective 
+immediately and does not need a restart. To do so, take the susbystem/module you are interested in and prepend
```
Comment (dedemorton) [inline docs/static/logging.asciidoc]: "and do not need a restart"

File: docs/static/logging.asciidoc [hunk 6]
```diff
@@ -0,0 +1,69 @@
+[[logging]]
+== Logstash's Internal Logging
```
Comment (dedemorton) [inline docs/static/logging.asciidoc]: I'd be inclined to just to call this topic "Logging" (saying "internal logging" might suggest to users that it's not something they can configure.)

Let me know where you want to add this section. I think it makes sense under "Setting Up and Running Logstash".

**Comment by dedemorton** [comment]
LGTM, other than a few minor typos. 

Tal and I spoke last week about having a section called "Logging API" that's at the same level as the Monitoring API, but maybe that's not necessary yet until there are more settings APIs? Let me know if you want me to add it.

**Comment by suyograo** [comment]
@dedemorton fixed typos. +1 on putting this under "Setting Up and Running Logstash".

**Comment by suyograo** [comment]
Agreed re: settings API. I added Logging APIs to this section itself, that could be sufficient for now until we have more settings.

**Pull request closed without merging**